Composer/packages/client/src/recoilModel/selectors/dialogs.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { DialogInfo } from '@bfc/shared';
+import { selectorFamily } from 'recoil';
+
+import { dialogIdsState, dialogState } from '../atoms';
+
+export const dialogsState = selectorFamily<DialogInfo[], string>({
+  key: 'dialogs',
+  get: (projectId: string) => ({ get }) => {
+    const dialogIds = get(dialogIdsState(projectId));
+
+    return dialogIds.map((dialogId) => {
+      return get(dialogState({ projectId, dialogId }));
+    });
+  },
+  set: (projectId: string) => ({ set }, newDialogs) => {
```

Comment:
Isn't newDialogs expected to be an array (according to Typescript)?  Do you need to defend / silently fail if someone passes a non-array?